cmd/anonymizer/app/writer/writer.go

```diff
@@ -130,7 +131,7 @@ func (w *Writer) WriteSpan(msg *model.Span) error {
 	if w.config.MaxSpansCount > 0 && w.spanCount >= w.config.MaxSpansCount {
 		w.logger.Info("Saved enough spans, exiting...")
 		w.Close()
-		os.Exit(0)
+		return errors.New("saved MaxSpansCount")
```

Comment:
* this needs to be a constant that can be checked externally
* you need to make changes to main.go that calls Write so that it checks for the error code and does the exit